test/gtest/uct/test_peer_failure.cc

```diff
@@ -280,32 +307,32 @@ UCS_TEST_SKIP_COND_P(test_uct_peer_failure, purge_failed_peer,
     send_recv_am(0);
     send_recv_am(1);
 
-    const ucs_time_t loop_end_limit = ucs::get_deadline();
     const size_t num_pend_sends     = 64ul;
-    const uint64_t send_data        = 0;
+    const ucs_time_t loop_end_limit = ucs::get_deadline();
     std::vector<pending_send_request_t> reqs(num_pend_sends);
 
     {
         scoped_log_handler slh(wrap_errors_logger);
-
         ucs_status_t status;
+
+        fill_resources(false, loop_end_limit);
+        kill_receiver();
+
```

Comment:
what use case is fixed here?

you killed the receiver, so your sends will not be completed even if async progress gets invoked.

Why need another loop above `fill_resources`, which already tries to get `NO_RESOURCE` until timeout?

Reply:
> you killed the receiver, so your sends will not be completed even if async progress gets invoked.

yes, they won't be completed, it send will not report the error until we don't make some CQ progress
so, it may happen if we are able to fill the resources until we really get the peer failure
otherwise, test will be skipped, since ERR_TIMEOUT is returned from AM Short